src/graphtypes/simplegraphs/simpledigraph.jl

```diff
@@ -116,6 +116,61 @@ function add_edge!(g::SimpleDiGraph, e::SimpleDiGraphEdge)
 end
 
 
+function add_edges!(g::SimpleDiGraph, es)
+    T = eltype(g)
+    nvnew = nvold = nv(g)
+    neold = ne(g)
+    ftouched = falses(nvold)
+    btouched = falses(nvold)
+
+    for e in es
+        s, d = T.(Tuple(e))
+        if s < 1 || d < 1
+            error("Nonpositive vertex number")
```

Comment:
A little more feedback would be helpful such as: `error("Nonpositive vertex number $s, $d")`

Reply:
I don't think we should be erroring at all, actually. `add_edge!` fails by returning `false`:
```
julia> add_edge!(g,-1,-3)
false
```

Reply:
"This is invalid input, ignore it" a bit untypical for julia.

Reply:
We did this because the try catch necessary to safely handle insertions was too slow. If adding an edge can raise an exception, then it needs to be in a try catch block.

Checking the return value is faster than handling the exception. And if you want to ignore failed edge insertions, then you just don't check it and everything works.

Reply:
Ok, that does not apply to this function

Reply:
It actually does, following our principle of "consistency". What I'd suggest (open to discussion) is returning a vector of booleans (or a bitvector) indicating the success of adding each of the edges.

Reply:
Putting to much faith into a single principle hinders more than it helps. `add_edges!` is not the same as multiple calls to `add_edge!`, starting with that `add_edges!` would become very unuseful if it would not implicitly add vertices, something `add_edge!` does not (and that is the whole point of not producing an error for illegal input). The function already informs if there were any edges added, negative vertex numbers are just an ordinary `IndexError`s.

Reply:
What about returning the number of edges added? Then the user can check 

```julia
numinsertions = add_edges!(g, newedges)
if numinsertions < length(newedges)
    # some edges were missing, find out why.
end
```

Reply:
> Putting too much faith into a single principle hinders more than it helps.

With respect, it's one of our three core tenets of LightGraphs (the others being "simple" and "performant") that have served us very well. I am not open to revisiting its utility.

> `add_edges!` is not the same as multiple calls to `add_edge!`

For one specific graph type, this may be true. But it may not be true for other graph types, nor is it consistent with `add_vertices!` vs `add_vertex!`.

> `add_edges!` would become very unuseful if it would not implicitly add vertices

This is one thing that bothers me. I'm leaning towards suggesting a different name for this function as a result.

Reply:
Just for a little context on my thinking:

I recently wrote some "production grade" python code and had to actually handle exceptions, unlike my interactive data analysis work where unhandled exceptions crash the program and no one cares. 

Basically every line of the the program is in a try/catch because every operation can raise an indexing error or something. And we are spending most of the time in the program on the "unhappy paths" of having caught at least one exception.
 
Recency bias is making me particularly sensitive to adding more exception handling to LG.

Reply:
> I recently wrote some "production grade" python code

Not sure whether this sentence throws a `DomainError` or a `SyntaxError`. :)

Reply:
I was thinking about `grow_graph!` or something along the lines. And I think you should reconsider your position about not throwing an error for invalid input (versus not throwing an error for input which is only invalid because of the context, like removing an edge twice).

Reply:
I am more inclined to agree that `add_edges!` could throw an exception because it is less likely to be used in a tight loop. But I think that matching the behavior of `add_vertices!` is important. We can always re-evaluate down the line if necessary.

```julia
"""
    add_vertices!(g, n)
Add `n` new vertices to the graph `g`.
Return `true` if all vertices were added successfully, `false` otherwise.
"""
add_vertices!(g::AbstractGraph, n::Integer) = all([add_vertex!(g) for i = 1:n])
```